File: trijectory/engine/engine_param.py
import dataclasses
from typing import Literal

from trijectory.type_aliases import ArrF64


@dataclasses.dataclass
class TrajectoryParam:
    max_time: float
    time_step: float
    log_rate: int
    escape_debounce_time: float
    min_distance: float
    method: Literal["euler", "rk22", "rk44"]
    mass: ArrF64 | None

File: src/rs_trijectory/engine_param.rs
use ndarray::Array1;
use std::str::FromStr;

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum Method {
    Euler,
    Rk22,
    Rk44,
}

impl FromStr for Method {
    type Err = ();

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s {
            "euler" => Ok(Method::Euler),
            "rk22" => Ok(Method::Rk22),
            "rk44" => Ok(Method::Rk44),
            _ => Err(()),
        }
    }
}

pub struct TrijectoryParam {
    pub max_time: f64,
    pub time_step: f64,
    pub escape_debounce_time: f64,
    pub min_distance: f64,
    pub method: Method,
    pub mass: Array1<f64>,
}
